EngineLayer/Psm.cs

```diff
@@ -16,10 +16,15 @@ public class Psm
         #region Private Fields
 
         private const double tolInDaForPreferringHavingMods = 0.03;
-        private Dictionary<CompactPeptide, Tuple<int, HashSet<PeptideWithSetModifications>>> compactPeptides = new Dictionary<CompactPeptide, Tuple<int, HashSet<PeptideWithSetModifications>>>();
 
         #endregion Private Fields
 
+        #region Public Fields
+
+        public Dictionary<CompactPeptideBase, Tuple<int, HashSet<PeptideWithSetModifications>>> compactPeptides = new Dictionary<CompactPeptideBase, Tuple<int, HashSet<PeptideWithSetModifications>>>();
```

Comment:
Line 392-3 of NSESTAP uses a TryGetValue  and Remove on the compactPeptides dictionary, which can't be conducted on a IEnumerable(KeyValuePairs). Also uses a ContainsKey (403) and needs an add function (405).